Makefile

```diff
@@ -32,6 +34,13 @@ PANDOC_PDF_OPTIONS= --pdf-engine=xelatex
 PANDOC_DOCX_OPTIONS=
 PANDOC_EPUB_OPTIONS=--to epub3
 
+# Check if the pandoc is installed
+check-pandoc:
+	@which pandoc > /dev/null || (echo "Pandoc not found, installing..." && make install-pandoc)
+
+install-pandoc:
+	sudo apt-get update && sudo apt-get install -y pandoc
```

Comment:
This target is specific to distributions that use the apt-get package manager, Debian and Ubuntu mainly. This excludes other distributions like Fedora and OpenSUSE and many more. And other operating systems like macOS and Windows.

I think it's better to fail with a warning that a certain program is necessary, installing things system-wide just for a single project can be undesirable. There's many ways to do this so I wouldn't choose any particular one for the user. I myself use the following to provide me with a project-specific environment using `nix-shell`:

```nix
{ pkgs ? import <nixpkgs> {} }:

let inherit (pkgs) mkShell; in

mkShell {
  packages = with pkgs; [ mkdocs pandoc texlive.combined.scheme-small ];
}
```